app/src/org/commcare/connect/ConnectManager.java

```diff
@@ -985,4 +986,11 @@ public static String generatePassword() {
 
         return password.toString();
     }
+
+    public static boolean shouldShowJobStatus(Activity activity) {
+        String appId = CommCareApplication.instance().getCurrentApp().getUniqueId();
+        ConnectAppRecord record = ConnectManager.getAppRecord(activity, appId);
+        ConnectJobRecord job = ConnectManager.getActiveJob();
+        return job.getStatus() == STATUS_DELIVERING && record.getIsLearning();
```

Comment:
For the logic here, we want to check for two possibilities where we want to show the Job Status button:
1. Job in learning state, logged in to learning app
2. Job in delivery state, logged in to delivery app

So (status == LEARNING && record.getIsLearning) || (status == DELIVERY && !record.getisLearning)